docs/api/http-v1-batch.md

```diff
@@ -191,11 +191,47 @@ between the requesting user and the repository.
 * 403 - The user has **read**, but not **write** access. Only applicable when
 the `operation` in the request is "upload."
 * 404 - The repository does not exist for the user.
+* 422 - Validation error with one or more of the objects in the request.
+
+Validation errors can only occur on "upload" requests. Servers should verify
+that OIDs are valid SHA-256 strings, and that sizes are positive integers.
+Servers may set an upper bound of allowed object sizes too.
 
-### Object Errors
+Here's an example of a response describing a validation error. Error responses
+must include a "message" property. They may include "request_id" and
+"documentation_url" properties to assist end users in getting help.
 
-The server can return specific errors for objects in the batch request. Here's
-a sample request and response that includes one missing object:
+< HTTP/1.1 422 Unprocessable Entity
+< Content-Type: application/vnd.git-lfs+json
+<
+< {
+<   "message": "Validation error",
+<   "documentation_url": "https://git-lfs-server.com/docs/errors",
+<   "request_id": "123",
+<   "objects": [
+<     {
+<       "oid": "1111111",
+<       "size": 123
+>     },
+<     {
+<       "oid": "2222222",
+<       "size": -1,
+<       "error": {
+<         "code": 422,
+<         "message": "Size is too low"
+<       }
+>     }
+<   ]
+< }
+
+The HTTP 422 response indicates that none of the objects were saved. The client
+may batch up the valid objects for another attempt at uploading them. However,
+clients should abort higher level operations like a Git push since not all of
+the objects were uploaded separately.
+
+It is possible for servers to respond with a 200, and just annotate specific
+objects that failed. Here's a sample request and response that includes one
+missing object:
```

Comment:
> IMO, a request to upload an existing file is not an error. The object should have no error and no actions set on it. The lack of an upload action means the client can skip it.

100% agree, when I said 'a similar response' I meant a missing upload entry for the object in the response, sorry for the ambiguity.

> I think in previous API versions, the server returned a download action on existing objects instead. I don't think the client even bothers to check this though.

Yeah, and I've mirrored that in test servers I've built too :) But that's because the old API didn't have an intended operation; leaving the object out seems more consistent.